packages/blockchain/test/reorg.ts

```diff
@@ -0,0 +1,86 @@
+import Common from '@ethereumjs/common'
+import { Block } from '@ethereumjs/block'
+import { BN } from 'ethereumjs-util'
+import tape from 'tape'
+import Blockchain from '../src'
+import { generateConsecutiveBlock } from './util'
+
+const genesis = Block.fromBlockData({
+  header: {
+    number: new BN(0),
+    difficulty: new BN(0x020000),
+    gasLimit: new BN(8000000),
+  },
+})
+
+tape('reorg tests', (t) => {
+  t.test(
+    'should correctly reorg the chain if the total difficulty is higher on a lower block number than the current head block',
+    async (st) => {
+      const common = new Common({ chain: 'mainnet', hardfork: 'muirGlacier' })
+      const blockchain = new Blockchain({ validateBlocks: true, validatePow: false, common })
+      await blockchain.putBlock(genesis)
+
+      const blocks_lowTD: Block[] = []
+      const blocks_highTD: Block[] = []
+
+      blocks_lowTD.push(generateConsecutiveBlock(genesis, 0))
+
+      const TD_Low = new BN(genesis.header.difficulty).add(
+        new BN(blocks_lowTD[0].header.difficulty)
+      )
+      const TD_High = new BN(genesis.header.difficulty)
```

Comment:
good stuff. `difficulty` and `number` should already be `BN`s now right? there's a lot of `new BN()`s here and below.

Reply:
Will resolve here, these (should be) resolved in #926